micro-rdk/src/common/restart_monitor.rs

```diff
@@ -0,0 +1,47 @@
+use super::app_client::{AppClient, AppClientError, PeriodicAppClientTask};
+use futures_lite::Future;
+use std::pin::Pin;
+use std::time::Duration;
+
+pub struct RestartMonitor {
+    restart_hook: Option<Box<dyn FnOnce()>>,
+}
+
+impl RestartMonitor {
+    pub fn new(restart_hook: impl FnOnce() + 'static) -> Self {
```

Comment:
I also would have liked to make `restart_hook` be `FnOnce -> !` but that doesn't seem to be supported yet.